containers/unit_tests/TestErrorReporter.hpp

```diff
@@ -86,10 +94,15 @@ void TestErrorReporter() {
   std::vector<typename tester_type::report_type> reports;
 
   tester_type test1(100, 10);
-  test1.m_errorReporter.getReports(reporters, reports);
+
+  std::tie(reporters, reports) = test1.m_errorReporter.get_reports();
   checkReportersAndReportsAgree(reporters, reports);
 
   tester_type test2(10, 100);
+  std::tie(reporters, reports) = test2.m_errorReporter.get_reports();
```

Comment:
Maybe have one of these be structured binding so that both expected usages are in the unit test.